gadget/gadget.go

```diff
@@ -256,22 +260,17 @@ func (v *VolumeStructure) IsRoleMBR() bool {
 	return v.Role == schemaMBR
 }
 
-// HasFilesystem returns true if the structure is using a filesystem.
-func (vs *VolumeStructure) HasFilesystem() bool {
-	return vs.Filesystem != "none" && vs.Filesystem != ""
-}
-
-// willHaveFilesystem considers also partially defined filesystem.
-func (vs *VolumeStructure) willHaveFilesystem(v *Volume) bool {
-	if vs.HasFilesystem() {
+// HasFilesystem returns true if the structure definition expects a filesystem.
+func (vs *VolumeStructure) HasFilesystem(v *Volume) bool {
```

Comment:
how annoying would be to introduce backpointers from VolumeStructure to Structure instead of this? it also make the change hard to follow atm, if we can't avoid this it needs to be done separately from using partial information as this is quite a churny change

Reply:
Thinking about this, maybe the cleaniest will be to have a new variable in the structure that tells us if the structure requires a filesystem or not, as in the end that is implicit information encoded in gadget.yaml. The variable can then be set when reading this file.

Reply:
that sounds fine assuming we don't have other potential methods on *Structure that will need to know partial content? Seems it should be done as a preparatory PR before this one